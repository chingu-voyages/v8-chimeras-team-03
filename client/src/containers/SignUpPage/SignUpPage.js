import React, { Component } from "react";
import { Link } from "react-router-dom";
import classes from "./SignUpPage.module.scss";

class SignUpPage extends Component {
  state = {
    email: "",
    password: "",
    repeatedPassword: "",
    userAgreedToTerms : false
  }

  handleCheckbox = () => {
    this.setState({userAgreedToTerms : !this.state.userAgreedToTerms});
  }
  handleEmail = (event) => {
    this.setState({ email: event.target.value });
  }
  handlePassword = (event) => {
    this.setState({ password: event.target.value });
  }
  handleConfirmPassword = (event) => {
    this.setState({ repeatedPassword: event.target.value });
  }
  handleSubmit = (event) => {
    if(!this.canBeSubmitted){
      event.preventDefault();
      return;
    }else{
    const {email, password, repeatedPassword, userAgreedToTerms} = this.state;
    alert(`Signed up with email: ${email} password: ${password} repeatedPassword: ${repeatedPassword} termsAgreed: ${userAgreedToTerms}`)
    }
  }
  canBeSubmitted = () => {
    const { email, password, repeatedPassword, userAgreedToTerms } = this.state;
    return (  email.length > 0 && 
              password.length > 0 && 
              repeatedPassword.length > 0 &&
              password === repeatedPassword && 
              userAgreedToTerms === true
          );
  }

  render() {
    const isEnabled = this.canBeSubmitted();
    return (
      <div className={classes.signup}>
        <h1>Let's get started</h1>
        <p>Kickstart your productivity with toggl clone</p>
        <form className={classes.form} onSubmit={this.handleSubmit}>
          <label htmlFor="email">EMAIL ADDRESS</label>
          <input type="email" id="email" onChange={this.handleEmail} value={this.state.email} />
          <label htmlFor="password">SET A PASSWORD</label>
          <input type="password" id="password" onChange={this.handlePassword} value={this.state.password}/>
          <label htmlFor="password">CONFIRM PASSWORD</label>
          <input type="password" id="confirmPassword" onChange={this.handleConfirmPassword} value={this.state.repeatedPassword}/>
          <div><input type="checkbox" onChange={this.handleCheckbox} id="checkbox" value={this.state.userAgreedToTerms} /> <p>Accept the terms</p></div>
          <button className={classes['sign-up-btn']} type="submit" disabled={!isEnabled}>
            SIGN UP FOR FREE <span />
          </button>
          <button className={classes['sign-up-google-btn']} >
            SIGN UP WITH GOOGLE
            <span />
          </button>
        </form>
        <p className={classes['no-acc']}>Already signed Up?</p>
        <button className={classes['sign-up-google-btn']}>
          <Link to="/login">
             Login here<span />
          </Link>
        </button>
      </div>
    )
  }
}

export default SignUpPage;
