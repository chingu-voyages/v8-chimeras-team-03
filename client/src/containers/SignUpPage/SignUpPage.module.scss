$grey: #e3e8eb;
$beige: #feeddb;
$red: #e1393f;

.background-grey {
  background-color: $grey;
  position: absolute;
  top: 368px;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
  overflow: visible;
  @media screen and (min-width: 768px) {
    top: 353px;
  }
  @media screen and (min-width: 1280px) {
    top: 476px;
  }
}

.signup {
  text-align: center;
  background-color: $beige;
  overflow: auto;
  min-height: 100vh;

  h1 {
    color: black;
    font-size: 30px;
    margin-top: 117px;
    @media screen and (min-width: 768px) {
      margin-top: 124px;
    }
    @media screen and (min-width: 1280px) {
      margin-top: 169px;
      font-size: 35px;
    }
  }
  p {
    font-size: 14px;
    opacity: 0.8;
    margin-top: 20px;
    color: black;
    @media screen and (min-width: 1280px) {
      font-size: 20px;
    }
  }
  .no-acc {
    z-index: 3;
    position: relative;
    color: black;
    font-size: 14px;
    opacity: 1;
    font-weight: bold;
    margin-top: 34px;
  }
  button {
    // last button on the page
    cursor: pointer;
    z-index: 3;
    position: relative;
    width: 225px;
    height: 33px;
    border: none;
    outline: none;
    border-radius: 17px;
    background-color: white;
    margin-top: 20px;
    @media screen and (min-width: 768px) {
      width: 353px;
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      margin-top: 39px;
      margin-bottom: 39px;
    }
    a {
      text-decoration: none;
      color: black;
      font-size: 14px;
      font-weight: bold;
      @media screen and (min-width: 1280px) {
        font-size: 20px;
      }
      span {
        width: 0;
        height: 0;
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        border-left: 5px solid black;
        display: inline-block;
        margin-left: 4px;
        @media screen and (min-width: 1280px) {
          border-top: 7px solid transparent;
          border-bottom: 7px solid transparent;
          border-left: 7px solid black;
        }
      }
    }
  }
}

.form {
  z-index: 2;
  position: relative;
  display: flex;
  flex-direction: column;
  width: 316px;
  height: 402px;
  background-color: white;
  margin: 0 auto;
  padding-left: 44px;
  padding-right: 44px;
  padding-top: 34px;
  border: 1px solid #707070;
  text-align: left;
  margin-top: 50px;
  @media screen and (min-width: 768px) {
    margin-top: 80px;
    width: 441px;
    height: 550px;
  }
  @media screen and (min-width: 1280px) {
    margin-top: 82px;
    width: 653px;
    height: 800px;
    padding: 50px 65px 0 65px;
  }
  label {
    font-size: 14px;
    font-weight: bold;
    @media screen and (min-width: 1280px) {
      font-size: 22px;
    }
  }
  input {
    background-color: $grey;
    width: 228px;
    height: 36px;
    outline: none;
    border: 1px solid #707070;
    padding-left: 10px;
    font-size: 13px;
    margin-top: 9px;
    margin-bottom: 9px;
    @media screen and (min-width: 768px) {
      width: 353px;
      margin-bottom: 34px;
    }
    @media screen and (min-width: 1280px) {
      width: 522px;
      height: 54px;
      font-size: 14px;
    }
  }
  div {
    // display: inline;
    // changed to display: flex
    display: flex;
    align-items: center;
    input {
      width: 30px;
      // changed mb
      @media screen and(min-width: 768px) {
        margin-bottom: 7px;
      }
    }
    p {
      display: inline;
      text-decoration: none;
      // added margins
      margin-left: 10px;
      @media screen and(min-width: 768px) {
        margin-left: 14px;
      }
      @media screen and(min-width: 1280px) {
        margin-left: 32px;
        font-size: 20px;
      }
    }
  }

  p {
    // forgot password
    font-size: 14px;
    text-decoration: underline;
    font-weight: bold;
    color: black;
    opacity: 1;
    margin-top: 0; // overwritten mt
    @media screen and (min-width: 768px) {
      // margin-top: 16px;
    }
  }
  .sign-up-btn {
    cursor: pointer;
    background-color: $red;
    border: none;
    outline: none;
    margin-top: 23px;
    width: 225px;
    height: 33px;
    border-radius: 17px;
    color: white;
    font-size: 14px;
    font-weight: bold;
    text-align: center;
    @media screen and (min-width: 768px) {
      width: 350px;
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      font-size: 20px;
      margin-top: 50px;
      margin-bottom: 29px;
    }
    span {
      width: 0;
      height: 0;
      border-top: 5px solid transparent;
      border-bottom: 5px solid transparent;
      border-left: 5px solid white;
      display: inline-block;
      margin-left: 4px;
      @media screen and (min-width: 1280px) {
        border-top: 7px solid transparent;
        border-bottom: 7px solid transparent;
        border-left: 7px solid white;
      }
    }
  }
  .sign-up-google-btn {
    cursor: pointer;
    border: 1px solid black;
    outline: none;
    width: 225px;
    height: 33px;
    border-radius: 17px;
    font-size: 14px;
    font-weight: bold;
    text-align: center;
    background-color: white;
    margin-top: 18px;
    ///// ////////////////////////
    margin-bottom: 32px;
    ///////////////////////////
    @media screen and (min-width: 768px) {
      width: 350px;
      //////////////
      margin-bottom: 50px;
      /////////////
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      font-size: 20px;
      margin-top: 47px;
      ////
      margin-bottom: 61px;
      ////
    }
    span {
      width: 0;
      height: 0;
      border-top: 5px solid transparent;
      border-bottom: 5px solid transparent;
      border-left: 5px solid black;
      display: inline-block;
      margin-left: 4px;
      @media screen and (min-width: 1280px) {
        border-top: 7px solid transparent;
        border-bottom: 7px solid transparent;
        border-left: 7px solid white;
      }
    }
  }
  .sign-up-btn:disabled {
    background-color: $grey;
  }
}
