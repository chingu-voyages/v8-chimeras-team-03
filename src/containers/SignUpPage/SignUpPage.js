import React, { Component } from "react";
import { Redirect, Link } from "react-router-dom";
import { auth } from "../../components/Firebase/firebase";
import classes from "./SignUpPage.module.scss";

class SignUpPage extends Component {
  state = {
      redirect: false,
      email: "",
      password: "",
      repeatedPassword: "",
      userAgreedToTerms : false
    }

  handleSignUp = async event => {
    if(!this.canBeSubmitted){
      event.preventDefault();
      return;
    }
    const { email, password } = this.state;
    try {
      await auth.createUserWithEmailAndPassword(email, password);
      this.setState({
        redirect: true
      });
    } catch (error) {
      alert(error);
    }
  }
  handleCheckbox = () => {
    this.setState({userAgreedToTerms : !this.state.userAgreedToTerms});
  }
  handleEmail = (event) => {
    this.setState({ email: event.target.value });
  }
  handlePassword = (event) => {
    this.setState({ password: event.target.value });
  }
  handleConfirmPassword = (event) => {
    this.setState({ repeatedPassword: event.target.value });
  }
  canBeSubmitted = () => {
    const { email, password, repeatedPassword, userAgreedToTerms } = this.state;
    return (  email.length > 0 && 
              password.length > 0 && 
              repeatedPassword.length > 0 &&
              password === repeatedPassword && 
              userAgreedToTerms === true
          );
  }
  
  render() {
    const isEnabled = this.canBeSubmitted();
    const { redirect } = this.state;
    if (redirect) {
      return <Redirect to="/"/>
    }
    return (
      <div className={classes.signup}>
        <h1>Let's get started</h1>
        <p>Kickstart your productivity with toggl clone</p>
        <form className={classes.form} onSubmit={this.handleSignUp}>
          <label htmlFor="email">EMAIL ADDRESS</label>
          <input type="email" id="email" onChange={this.handleEmail} value={this.state.email} />
          <label htmlFor="password">SET A PASSWORD</label>
          <input type="password" id="password" onChange={this.handlePassword} value={this.state.password}/>
          <label htmlFor="password">CONFIRM PASSWORD</label>
          <input type="password" id="confirmPassword" onChange={this.handleConfirmPassword} value={this.state.repeatedPassword}/>
          <div><input type="checkbox" onChange={this.handleCheckbox} id="checkbox" value={this.state.userAgreedToTerms} /> <p>Accept the terms</p></div>
          <button className={classes['sign-up-btn']} type="submit" disabled={!isEnabled}>
            SIGN UP FOR FREE <span />
          </button>
          <button className={classes['sign-up-google-btn']} >
            SIGN UP WITH GOOGLE
            <span />
          </button>
        </form>
        <p className={classes['no-acc']}>Already signed Up?</p>
        <button className={classes['sign-up-google-btn']}>
          <Link to="/login">
             Login here<span />
          </Link>
        </button>
      </div>
    )
  }
}
export default SignUpPage;
