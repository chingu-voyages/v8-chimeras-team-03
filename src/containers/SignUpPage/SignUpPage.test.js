import React from "react";
import SignUpPage from "./SignUpPage";
import { shallow } from "enzyme";

it("renders <SingUpPage/>", () => {
  const wrapper = shallow(<SignUpPage />);
  expect(wrapper.text("SignUp Page")).toEqual("SignUp Page");
});

it("should render without crashing", () => {
  const wrapper = shallow(<SignUpPage />);
  expect(wrapper).toMatchSnapshot();
});
