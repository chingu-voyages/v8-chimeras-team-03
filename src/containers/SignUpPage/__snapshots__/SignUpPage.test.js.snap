// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "SignUp Page",
      "className": "x",
    },
    "ref": null,
    "rendered": "SignUp Page",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "SignUp Page",
        "className": "x",
      },
      "ref": null,
      "rendered": "SignUp Page",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
